tensorflow/core/platform/file_system.h

```diff
@@ -68,9 +68,14 @@ class FileSystem {
   /// The ownership of the returned RandomAccessFile is passed to the caller
   /// and the object should be deleted when is not used.
   virtual tensorflow::Status NewRandomAccessFile(
-      const string& fname,
-      std::unique_ptr<RandomAccessFile>*
-          result /*, TransactionToken* token = nullptr */) = 0;
+      const string& fname, std::unique_ptr<RandomAccessFile>* result) = 0;
+
+  virtual tensorflow::Status NewRandomAccessFile(
+      const string& fname, std::unique_ptr<RandomAccessFile>* result,
+      TransactionToken* token){
+      // We duplicate these methods due to Google internal coding style prevents
+      // virtual functions with default arguments. See PR #41615.
+  };
```

Comment:
I'm wondering if there is something to prevent us from doing

```suggestion
      const string& fname, std::unique_ptr<RandomAccessFile>* result) {
      return NewRandomAccessFile(fname, nullptr, result);
  }

  virtual tensorflow::Status NewRandomAccessFile(
      const string& fname, TransactionToken* token,
      std::unique_ptr<RandomAccessFile>* result) = 0;
```

This way we have the old methods redirect to the new ones by default and the new ones still use the style of having output parameters last.

Reply:
Yes that is the idea. But if I do that now I will have to change everything at once. I am trying to split it into small chunks so reviewing is simpler.

Reply:
But the newly introduced methods could already have the arguments in the right order, right?

Reply:
Sure that is correct. Let me fix it.

Reply:
@mihaimaruseac since you approved this. Do you want me to fix them in a subsequent PR or on this one?

Reply:
Upcoming one should be ok